core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1054,11 +1054,37 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     case "INTERVAL_SECONDS":
       unparseIntervalSeconds(writer, call, leftPrec, rightPrec);
       break;
+    case "PARSE_DATE":
+    case "PARSE_TIME":
+      unparseDateTime(writer, call, leftPrec, rightPrec);
+      break;
     default:
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
 
+  protected void unparseDateTime(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlCall formatCall;
+    switch (call.getOperator().getName()) {
+    case "PARSE_DATE":
+    case "PARSE_TIME":
+      String dateFormat = call.operand(0) instanceof SqlCharStringLiteral
+          ? ((NlsString) requireNonNull(((SqlCharStringLiteral) call.operand(0)).getValue()))
+          .getValue() : call.operand(0).toString();
+      SqlOperator function = call.getOperator();
+      if (!dateFormat.contains("%")) {
+        formatCall = function.createCall(SqlParserPos.ZERO,
+            createDateTimeFormatSqlCharLiteral(dateFormat), call.operand(1));
+        function.unparse(writer, formatCall, leftPrec, rightPrec);
+      } else {
+        function.unparse(writer, call, leftPrec, rightPrec);
+      }
+      break;
+    default:
```

Comment:
switch is not required as it will never go into default case because in `unparseOtherFunction` you already have a switch in which you are calling `unparseDateTime` only for "PARSE_DATE" and "PARSE_TIME" .
So there will be no default case in  `unparseDateTime`.